single_stage_detector/README.md

```diff
@@ -0,0 +1,293 @@
+objects, along with their classes. For example:
+
+![](https://upload.wikimedia.org/wikipedia/commons/3/38/Detected-with-YOLO--Schreibtisch-mit-Objekten.jpg)
+
+SSD is a one-stage detector, both localization and classification are done in a
+single pass of the network. This allows for a faster inference than region
+proposal network (RPN) based networks, making it more suited for real time
+applications like automotive and low power devices like mobile phones. This is
+also sometimes referred to as being a "single shot" detector for inference.
+
+# Running the reference
+The reference code is intended to run on NVIDIA GPUs, it is tested on A100 and
+V100 cards but other GPUs should work too.
+
+## Prerequisites
+The recommended way to run the reference is with docker. You need to setup your
+machine with:
+1. NVIDIA drivers. Can be installed with [CUDA toolkit](https://developer.nvidia.com/cuda-downloads).
+2. [Docker](https://docs.docker.com/engine/install/)
+3. [NVIDIA container runtime](https://github.com/NVIDIA/nvidia-docker)
+
+If you are using Ubuntu 20.04, you can install these components by running:
+```bash
+bash ./install_cuda_docker.sh
+```
+
+In addition to the above requirements, it's recommended to use
+[Slurm](https://developer.nvidia.com/slurm) for Multi-node training
+(the installation and configuration of Slurm is cluster depended and not
+covered in this README).
+
+## Building the docker image
+Once the prerequisites are met, you can build the benchmark docker image with:
+```bash
+cd ./single_stage_detector/
+docker build -t mlperf/single_stage_detector .
+```
+
+## Download dataset
+The benchmark uses a subset of [OpenImages-v6](https://storage.googleapis.com/openimages/web/index.html).
+To download the subset:
+```bash
+cd ./single_stage_detector/scripts
+pip install fiftyone
+./download_openimages_mlperf.sh -d <DOWNLOAD_PATH>
+```
+
+The script will download the benchmark subset with metadata and labels, then
+convert the labels to [COCO](https://cocodataset.org/#home) format. The
+downloaded dataset size is 352GB and the expected folder structure after
+running the script is:
+```
+<DOWNLOAD_PATH>
+│
+└───info.json
+│
+└───train
+│   └─── data
+│   │      000002b66c9c498e.jpg
+│   │      000002b97e5471a0.jpg
+│   │      ...
+│   └─── metadata
+│   │      classes.csv
+│   │      hierarchy.json
+│   │      image_ids.csv
+│   └─── labels
+│          detections.csv
+│          openimages-mlperf.json
+│
+└───validation
+    └─── data
+    │      0001eeaf4aed83f9.jpg
+    │      0004886b7d043cfd.jpg
+    │      ...
+    └─── metadata
+    │      classes.csv
+    │      hierarchy.json
+    │      image_ids.csv
+    └─── labels
+           detections.csv
+           openimages-mlperf.json
+```
+
+Read more about the mlperf subset [here](#the-mlperf-subset).
+
+
+## Download the pretrained backbone
+The benchmark uses a ResNeXt50_32x4d backbone pretrained on ImageNet. The
```

Comment:
Out of curiosity (nothing to do here for this round), do we know how much faster we can converge if we pretrain the backbone on OpenImages?

Reply:
Unfortunately, we didn't do this experiment, but this is an interesting one, especially if we plan to move away from ImageNet for the classification benchmark.